msal4j-brokers/src/main/java/com/microsoft/aad/msal4jbrokers/MsalRuntimeBroker.java

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4jbrokers;
+
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+import com.microsoft.aad.msal4j.IBroker;
+import com.microsoft.aad.msal4j.InteractiveRequestParameters;
+import com.microsoft.aad.msal4j.PublicClientApplication;
+import com.microsoft.aad.msal4j.SilentParameters;
+import com.microsoft.aad.msal4j.UserNamePasswordParameters;
+import com.microsoft.aad.msal4j.MsalException;
+import com.microsoft.aad.msal4j.AuthenticationErrorCode;
+import com.microsoft.aad.msal4j.IAccount;
+import com.microsoft.azure.javamsalruntime.Account;
+import com.microsoft.azure.javamsalruntime.AuthParameters;
+import com.microsoft.azure.javamsalruntime.AuthResult;
+import com.microsoft.azure.javamsalruntime.MsalInteropException;
+import com.microsoft.azure.javamsalruntime.MsalRuntimeInterop;
+import com.microsoft.azure.javamsalruntime.ReadAccountResult;
+
+import java.net.MalformedURLException;
+import java.util.Collections;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+public class MsalRuntimeBroker implements IBroker {
+    private static MsalRuntimeInterop interop;
+
+    static {
+        try {
+            //MsalRuntimeInterop performs various initialization steps in a similar static block,
+            // so when an MsalRuntimeBroker is created this will cause the interop layer to initialize
+            interop = new MsalRuntimeInterop();
```

Comment:
Where is the logic to determine which flavor of msalruntime to use? Is it in the java interop layer?

Reply:
Yes, interop will handle 100% of any platform-specific stuff, and it choses the correct MSALRuntime dll here: https://github.com/AzureAD/microsoft-authentication-library-for-cpp/blob/9534b41ac5a168641c80f3ecfc97d9a6e765da19/msalruntime/interop/java/src/main/java/com/microsoft/azure/javamsalruntime/MsalRuntimeInterop.java#L252

The exact dll that gets loaded doesn't matter to the broker, it's all just the same Java object with the same set of methods either way (JNA handles actually loading and calling the methods in the dll)

Reply:
Thanks. Looking at this class MsalRuntimeInterop.java, I see a lot of TODOs. Lets make sure they are taken care of before we move on to the next phase of implementing logging and tracing.